cosmwasm/enclaves/shared/block-verifier/src/verify/registration.rs

```diff
@@ -0,0 +1,43 @@
+use crate::VERIFIED_MESSAGES;
+use log::{debug, error};
+use protobuf::Message;
+
+pub fn verify_reg_msg(certificate: &[u8]) -> bool {
+    let mut verified_msgs = VERIFIED_MESSAGES.lock().unwrap();
+    let next = verified_msgs.get_next();
+
+    let result = if let Some(msg) = next {
+        // I assume @assaf will add the message type to the verified messages, so I'm going to leave this here for now
+        // without a check - it will fail in the parse either way
+
+        // if !check_message_is_reg(&msg) {
+        //     error!("Error failed to validate registration message - 0x7535");
+        //     false
+        // }
+
+        match cosmos_proto::registration::v1beta1::msg::RaAuthenticate::parse_from_bytes(&msg) {
+            Ok(ra_msg) => {
+                if ra_msg.certificate == certificate {
+                    return true;
+                }
+                error!("Error failed to validate registration message - 0x7535");
+                false
+            }
+            Err(e) => {
+                debug!("Error decoding registation protobuf: {}", e);
+                error!("Error decoding msg from block validator - 0xA0F2");
```

Comment:
What's the deal with this hex `0xA0F2`?

Reply:
I kind of like adding random error codes - makes finding the exact error easier in a month or two when someone randomly pastes their error. Also we can one day create a codex of error codes & their meaning/more info, etc.